app/src/main/java/tech/ula/ServerService.kt

```diff
@@ -244,6 +246,18 @@ class ServerService : Service() {
             sessionController.findAppSession(app.name, serviceType, appsFilesystem, sessionDao)
         }
 
+        if (!username.isEmpty()) {
```

Comment:
These checks / variable adjustments could be moved into the session controller so that they can be unit tested.